server/src/main/java/org/elasticsearch/index/store/LuceneFilesExtensions.java

```diff
@@ -96,11 +96,11 @@ public enum LuceneFilesExtensions {
      * that checks that all encountered file extensions are known to this class.
      * In the future, we would like to add a proper plugin extension point for this.
      */
-    @SuppressForbidden(
-        reason = "TODO Deprecate any lenient usage of Boolean#parseBoolean https://github.com/elastic/elasticsearch/issues/128993"
-    )
     private static boolean allowUnknownLuceneFileExtensions() {
-        return Boolean.parseBoolean(System.getProperty("es.allow_unknown_lucene_file_extensions", "false"));
+        return LenientBooleans.parseAndCheckForDeprecatedUsage(
```

Comment:
besides the category, we absolutely want to specify the name of the sys property